variables.tf

```diff
@@ -484,6 +484,32 @@ variable "bitbucket_elasticsearch_replicas" {
   default     = 2
 }
 
+variable "bitbucket_db_snapshot_identifier" {
+  description = "The identifier for the DB snapshot to restore from. The snapshot should be in the same AWS region as the DB instance."
+  default     = null
+  type        = string
+}
+
+variable "bitbucket_db_master_username" {
+  description = "Master username for the Bitbucket RDS instance."
+  type        = string
+  default     = null
+  validation {
+    condition     = can(regex("^[a-zA-Z_]([a-zA-Z0-9_]).{5,30}$", var.bitbucket_db_master_username)) || var.bitbucket_db_master_username == null
+    error_message = "Master username must be set. It must be between 6 and 31 characters long and start with a letter/underscore and contain combination of numbers, letters, and underscore."
+  }
+}
+
+variable "bitbucket_db_master_password" {
+  description = "Master password for the Bitbucket RDS instance."
+  type        = string
+  default     = null
+  validation {
```

Comment:
Instead of repeating the same validation at product levels, I think it's better to do the validation within RDS module. I added this to the [Confluence PR](https://github.com/atlassian-labs/data-center-terraform/pull/199/files).

Reply:
We add the validation in root level for the reason as a pattern. The root level has more visibility to users and set directly by config file.